pkg/skaffold/debug/debug.go

```diff
@@ -67,10 +67,8 @@ func applyDebuggingTransforms(l kubectl.ManifestList, retriever configurationRet
 	for _, manifest := range l {
 		obj, _, err := decodeFromYaml(manifest, nil, nil)
 		if err != nil {
```

Comment:
we need to be more specific to match the `no kind XYZ is registered for version ABC in scheme ` type of errors

Reply:
I'll respectfully disagree: any such validation should happen elsewhere.  `debug`'s current behaviour is to tweak the workload resources that it knows about, and skip any other resources.  We should extend that skipping behaviour to any resources that we (or our libraries) are unable to interpret too.

Reply:
I think what I meant is that there should be 3 branches: 

1) `err ~= no kind XYZ is registered for version ABC in scheme` -> continue 
2) `err != nil` but it's something else than unregistered schema -> return error 
3) ` err == nil` transform. 

WDYT?

Reply:
Hm...on second thought - my thinking was that  problem is that we don't know what the error is - if it is a valid yaml but with unknown kind - great let's skip it. if it's an invalid yaml though, I don't want to skip it. -- but you are right, if we want to do upfront YAML validation, it could/should be elsewhere, not necessarily in debug.